gnomad_qc/v4/create_release/calculate_variant_statistics.py

```diff
@@ -24,188 +51,204 @@
     datefmt="%m/%d/%Y %I:%M:%S %p",
 )
-logger = logging.getLogger("release")
+logger = logging.getLogger("per_sample_stats")
 logger.setLevel(logging.INFO)
 
 
-def create_per_sample_counts(
-    data_type: str = "exomes",
-    test: bool = False,
-    overwrite: bool = False,
+def create_per_sample_counts_ht(
+    mt: hl.MatrixTable,
+    annotation_ht: hl.Table,
     pass_filters: bool = False,
-    ukb_regions: bool = False,
-    broad_regions: bool = False,
+    ukb_capture: bool = False,
+    broad_capture: bool = False,
     by_csqs: bool = False,
     rare_variants: bool = False,
     vep_canonical: bool = True,
     vep_mane: bool = False,
 ) -> hl.Table:
     """
-    Write out Hail's sample_qc output for desired samples and variants.
-
-    Useful for finding the amount of variants (total or specific types) per sample.
-    Also prints distribution of total variants called per sample.
-
-    :param data_type: String of either "exomes" or "genomes" for the sample type.
-    :param test: Boolean for if you would like to use a small test set.
-    :param overwrite: Boolean to overwrite checkpoint files if requested.
-    :param pass_filters: Stratify by variants which pass all variant qc filters.
-    :param ukb_regions: Stratify by variants in UKB regions.
-    :param broad_regions: Stratify by variants in Broad regions.
-    :param by_csqs: Stratify by variant consequence.
-    :param rare_variants: Stratify by variants which have adj AF <0.1%.
-    :param vep_canonical: Stratify by canonical transcripts. If trying to get
-        variants in all transcripts, set it to False. Default is True.
-    :param vep_mane: Stratify by MANE transcripts. Default is False.
+    Create Table of Hail's sample_qc output broken down by requested variant groupings.
+
+    Useful for finding the number of variants (total or specific types) per sample.
+
+    The following per-sample variant counts are calculated:
+
+        - Total number of variants
+        - Number of variants that pass all variant qc filters
+        - Number of variants in UK Biobank capture regions
+        - Number of variants in Broad capture regions
+        - Number of rare variants (adj AF <0.1%)
+        - Number of loss-of-function variants
+        - Number of missense variants
+        - Number of synonymous variants
+        - Number of singletons
+        - Number of heterozygous variants
+        - Number of homozygous variants
+        - Number of non-reference variants
+        - Number of variants in UK Biobank and Broad capture regions
+
+    Additionally, counts can be computed for the following variant filtering criteria:
+
+        - Variants that pass all variant qc filters
+        - Variants in UK Biobank capture regions
+        - Variants in Broad capture regions
+        - Variants in UK Biobank and Broad capture regions
+        - Rare variants (adj AF <0.1%)
+        - Loss-of-function variants
```

Comment:
Reduant to Variant consequences?

Reply:
yeah, I didn't spend much time on this, just sort of added to give Dan the basic idea of what I'm looking for